flink-runtime/src/main/java/org/apache/flink/runtime/io/network/netty/NettyMessage.java

```diff
@@ -220,6 +220,10 @@ void releaseBuffer() {
 
 		@Override
 		ByteBuf write(ByteBufAllocator allocator) throws IOException {
+			if (null == buffer) {
+				throw new NullPointerException();
```

Comment:
actually, better use `Preconditions#checkNotNull(T)` - an additional error message there usually does not make too much sense as the failure is clear